pkg/controllers/corecluster/corecluster_controller.go

```diff
@@ -81,36 +81,49 @@ func (r *CoreClusterController) SetupWithManager(mgr ctrl.Manager) error {
 }
 
 // Reconcile reconciles the core cluster object for the openshift-cluster-api namespace.
-func (r *CoreClusterController) Reconcile(ctx context.Context, req reconcile.Request) (ctrl.Result, error) {
+func (r *CoreClusterController) Reconcile(ctx context.Context, req reconcile.Request) (_ ctrl.Result, reterror error) {
 	logger := log.FromContext(ctx).WithName(controllerName)
 	logger.Info("Reconciling core cluster")
 	defer logger.Info("Finished reconciling core cluster")
 
+	var (
+		err        error
+		conditions []configv1.ClusterOperatorStatusCondition
+	)
+
+	defer func() {
+		// This function runs before every return, to ensure the controller conditions
+		// for that codepath are correctly set to the ClusterOperator status.
+		if err := r.setControllerConditions(ctx, conditions); err != nil && reterror == nil {
+			reterror = fmt.Errorf("failed to sync cluster operator status: %w", err)
+		}
```

Comment:
If that status sync fails and there was another error, do we still want to log the error?

Reply:
This is bubbled up as we set `reterror` as a named return argument, so that would throw a reconciler error and log that in the controller logs.

Reply:
The original error gets bubbled up, looking at the logic as it is now, the status sync error would not get bubbled up

Reply:
Yes, I think you are right, the outer wrapped error takes precedence vs the named return argument error.
I logged the innermost one in case both happen.